templates/new_service/terraform/application/terraform.tf

```diff
@@ -0,0 +1,41 @@
+terraform {
+  required_version = "= 1.5.1"
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "3.62.1"
+    }
+    kubernetes = {
+      source  = "hashicorp/kubernetes"
+      version = "2.21.1"
+    }
+    statuscake = {
+      source  = "StatusCakeDev/statuscake"
+      version = "2.1.0"
+    }
```

Comment:
We don't need to specify these unless we're using them directly as we've got the versions specified in the modules themselves. This avoids us having to maintain two sets of versions.

Reply:
This got me thinking about the way we specify the versions in the modules and I think they're too unconstrained. I've opened a PR to use the `~>` operator to avoid breakages when a major version is released.

https://github.com/DFE-Digital/terraform-modules/pull/58

Reply:
By experience, pinning the version in the calling module as explicitly as possible avoids surprises. Otherwise you may end up with 2 slightly different provider versions between 2 users, or 2 environments. In the template, this would be used as a guide of the latest and greatest working configuration.

As discussed in https://github.com/DFE-Digital/terraform-modules/pull/58#issuecomment-1625248117 , ideally the module constraint is large and is only to prevent incompatibilities.

Reply:
As discussed, we've decided to stick with having the versions specified in the caller.